webnn/resources/test_data/reduce_sum.json

```diff
@@ -0,0 +1,782 @@
+{
+  "tests": [
+    {
+      "name": "reduceSum float32 1D tensor default options",
+      "inputs": {
+        "input": {
+          "shape": [24],
+          "data": [
+            -23.116038894167843,
+            12.95644885637492,
+            -66.95244678349574,
+            49.25963932971541,
+            69.76352446225414,
+            30.585307013632246,
+            -10.322283521403946,
+            96.27798039714031,
+            31.646707789509435,
+            50.44685780881761,
+            46.23992026181861,
+            3.438425050635672,
+            -29.153671310761723,
+            -71.61827137599674,
+            29.915138398194983,
+            -55.043894463599855,
+            12.643124156979283,
+            -36.256290208967236,
+            8.350975349272986,
+            15.984685655817941,
+            -22.18552852295987,
+            -45.433313937995614,
+            20.781868403551314,
+            -66.96646315625907
+          ],
```

Comment:
You'll have to be careful summing mixed signs, due to subtraction of nearly equal numbers magnifying error. For reduceSum, I think it wiser to curate the input tensors to all positive values, all negative values, and/or all integers. Otherwise you'll chase down errors. The ULP values given are with this restriction in mind, as there are no reasonable tolerance values for nearly equal numbers - you either have to relax them, which weakens the overall tolerance of all other cases, or you have to cross your fingers that you won't hit such a case (which is tough given the randomness of the values and the fact they can be added in any order depending on the implementation).

Reply:
Thanks. I will add such tests as your suggested.

Reply:
Done, added such tests for `reduceMean` and `reduceSum` ops, please have another look, thanks.

Reply:
Good that you added *more* test cases, but the *existing* problematic test cases of mixed signs in ReduceMean and ReduceSum remain and can encounter tolerance issues. So I was hoping those would be converted to consistent signs (or in light of the new test cases, be deleted).

```
      "name": "reduceMean float32 4D tensor default options",
      "inputs": {
        "input": {
          "shape": [2, 2, 2, 3],
          "data": [
            -1.913036787607325,
            95.38300711992477,
            -82.72358015843713,
            -32.94943032326678,
            91.42181191536278,
            -75.1219094678544,
            -70.81411336302938,
```

Reply:
Updated. Thanks.